src/main/java/gateways/Tags/TagDatabaseGateway.java

```diff
@@ -0,0 +1,103 @@
+package gateways.Tags;
+
+import entities.Tag;
+import entities.User;
+import gateways.DatabaseGateway;
+import gateways.UserDatabaseGateway;
+import services.CurrentUserService;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TagDatabaseGateway extends DatabaseGateway implements TagDataAccessInterface {
+
+    private final CurrentUserService currentUserService;
+    private UserDatabaseGateway userDatabaseGateway;
+    private List<User> userList;
+    private User currentUser;
+    private ArrayList<Tag> tagList;
+
+
+    /**
+     * Instantiating DatabaseGateway with a relative path will store the absolute path to the respective database.
+     * If the file at the path does not exist, one will be created
+     * This absolute path can be fetched using the getAbsoluteFilepath method
+     *
+     * @throws IOException Throws exception if encountering failed or interrupted IO exceptions
+     */
+    public TagDatabaseGateway(CurrentUserService currentUserService, UserDatabaseGateway userDatabaseGateway) throws IOException {
+        super(userDatabaseGateway.getAbsoluteFilepath());
+        this.currentUserService = currentUserService;
+        this.userDatabaseGateway = userDatabaseGateway;
+        load();
+    }
+
+    public void load() {
+        this.userList = userDatabaseGateway.loadFromFile();
```

Comment:
You should use getAll, loadFromFile shouldn't be called outside the class. I should have made it private, sorry about that.